bin/vivo_import_turtle.sh

```diff
@@ -1,14 +1,67 @@
 #!/bin/bash
 
-# Run this script from the cap-vivo-mapper app root, which should be installed under the user's home dir
-# usage: USR=joe@stanford.edu PASSWD=secret ./bin/vivo_import_turtle.sh | tee -ai ./log/vivo_import_turtle.log
+usage () {
+    cat <<HELP
 
-dtime=`date +%Y%m%d%H%S`
+$0 -u USER -p PASSWD -g GRAPH_URI -s SPARQL_URI -d DATA_PATH
 
-for f in /tmp/cap_vivo_rdf/*.ttl; do
-  echo "update=LOAD <file://$f> into graph <https://sul-vitro-dev.stanford.edu/default/vitro-kb-2>" > tmp.sparql
-  curl --insecure -d "email=$USR" -d "password=$PASSWD" -d '@tmp.sparql' https://sul-vivo-dev.stanford.edu/vivo/api/sparqlUpdate
+-u USER        : user (email) for access to VIVO SPARQL_URI
+-p PASSWD      : user password for access to VIVO SPARQL_URI
+-g GRAPH_URI   : VIVO graph identifier, e.g.
+                 https://{host}/default/vitro-kb-1
+-s SPARQL_URI  : VIVO API for SPARQL Update, e.g.
+                 https://{host}/vivo/api/sparqlUpdate
+-d DATA_PATH   : Directory containing turtle data files (*.ttl)
+
+HELP
+}
+
+# ---
+# Parse the command line options
+
+while getopts hu:p:g:s:d: opt; do
+    case $opt in
+    h)
+        usage
+        exit
+        ;;
+    u)
+        EMAIL=$OPTARG
+        ;;
+    p)
+        PASSWD=$OPTARG
+        ;;
+    g)
+        GRAPH_URI=$OPTARG
+        ;;
+    s)
+        SPARQL_URI=$OPTARG
+        ;;
+    d)
+        DATA_PATH=$OPTARG
+        ;;
+    \?)
+        echo "Invalid option: -$OPTARG" >&2
+        exit 1
+        ;;
+    :)
+        echo "Option -$OPTARG requires an argument." >&2
+        exit 1
+        ;;
+    esac
 done
 
-rm tmp.sparql
-mv ./log/vivo_import_turtle.log ./log/vivo_import_turtle.log.$dtime
+# ---
+# Function to load a file into a graph using SPARQL update
+
+sparql_update () {
+    file=$1
+    sparql="update=LOAD <file://${file}> into graph <${GRAPH_URI}>"
+    curl --insecure -d "$sparql" -d "email=$EMAIL" -d "password=$PASSWD" ${SPARQL_URI}
```

Comment:
@darrenleeweber did you test this to make sure that it works? I seem to remember that the `-d` for the sparql might require an actual sparql file. I don't think I would have chose to use a file otherwise. Is that why you have `$TMP_SPARQL` commented out, just in case?